internal/backend/runtime/omni/controllers/omni/infra_machine.go

```diff
@@ -108,7 +116,9 @@ func (h *infraMachineControllerHelper) transformExtraOutput(ctx context.Context,
 		return xerrors.NewTaggedf[qtransform.SkipReconcileTag]("the link is not created by a static infra provider")
 	}
 
-	if err = h.applyInfraMachineConfig(infraMachine, config); err != nil {
+	machineInfoCollected := machineStatus != nil && machineStatus.TypedSpec().Value.SecureBootStatus != nil
```

Comment:
I don't quite get this, not all machines would boot with secureboot, so why check this?

Reply:
It's just an indicator that machine info was collected. This struct has an `Enabled` field that is false if we collected the machine info and found out that it's not a secure boot machine. So, if this struct is not nil, we know that we collected the info.

I could have checked one of the other fields instead and it would have worked, but I wanted specifically to make sure this check would pass here in this controller: https://github.com/siderolabs/omni/blob/main/internal/backend/runtime/omni/controllers/omni/schematic_configuration.go#L166-L169

So I went with the exact same field.

Reply:
feels weird to check for that status instead of something like hardware info, maybe some historical reason.